### PR TITLE
Add capabilities to @Authenticated

### PR DESCRIPTION
**Pull request by ryanhall07**
This is the first part of https://github.com/cashapp/misk/issues/1078 to
add a new capabilities field to @Authenticated that will eventually
replace the roles field.

I did not Deprecate the roles field, because many repos have -WError set
which will fail the build if there is a Deprecated Warning.

### CODE REVIEW

File: misk/src/main/kotlin/misk/MiskCaller.kt [hunk 1]
```diff
@@ -9,7 +9,11 @@ data class MiskCaller(
   val user: String? = null,
 
   /** Set of roles to which the human user belongs, typically provided by the SSO infrastructure */
-  val roles: Set<String> = setOf()
+  // TODO(rhall): Deprecate this https://github.com/cashapp/misk/issues/1078
```
Comment (ryanhall07) [inline misk/src/main/kotlin/misk/MiskCaller.kt]: it's kind of lame that you can add `@Deprecated` with some kind of INFO level that won't break builds that have `-WError` set

File: misk/src/main/kotlin/misk/web/WebTab.kt [hunk 2]
```diff
@@ -26,6 +26,8 @@ class DashboardTab(
   url_path_prefix: String,
   val name: String,
   val category: String = "Container Admin",
+    // TODO(rhall): Deprecate this https://github.com/cashapp/misk/issues/1078
   roles: Set<String> = setOf(),
```
Comment (ryanhall07) [inline misk/src/main/kotlin/misk/web/WebTab.kt]: are these roles providing any value? couldn't we just show a 403 if one of the backend calls failed with Unauthorized? Is it just to gate showing the Tab? Seems not worth it to me at the expense of maintaining multiple ACL sets.

File: misk/src/main/kotlin/misk/MiskCaller.kt (hunk 1 above)
Comment (swankjesse) [reply misk/src/main/kotlin/misk/MiskCaller.kt]: I think `-WError` sucks

File: misk/src/main/kotlin/misk/web/WebTab.kt (hunk 2 above)
Comment (swankjesse) [reply misk/src/main/kotlin/misk/web/WebTab.kt]: Yeah, that might be simpler.

**Pull request merged**